### PR TITLE
Auto-refresh

### PR DESCRIPTION
**Pull request by ryanblock**
Per [conversation from auto-hydration](https://github.com/architect/sandbox/pull/3), some use cases sandbox ought to accommodate. Thus far this PR covers:
- Running a live `sandbox` and add a new function to `.arc` – `sandbox` should mount (and hydrate) the new function
- Running a live `sandbox` and remove a function from `.arc` – `sandbox` should unmount that function

Things not currently covered here / in scope (yet):
- Issues in `.arc` that would prevent `sandbox` from executing the function (eg folder missing, etc.) – `sandbox` could do some basic detection up front and present a warning
  -   To do so would necessitate kicking up the parser and validator
- Only refreshes `@http`; not (yet) inspecting and diffing against the actual content changes to the arc file
  - Example: if you make a change to `@tables`, it won't refresh `dynalite` and reflect those new changes until you fully restart `sandbox`
  - To do so would necessitate kicking up the parser, validator, and keeping some kind of before/after state of arc, as well as diffing logic and tasking various restarts

### CODE REVIEW

File: src/cli.js [hunk 1]
```diff
@@ -1,14 +1,64 @@
...
-  }
-  if (err) {
-    console.log(err)
-    process.exit(1)
-  }
-})
+
+start()
+function start() {
+  sandbox.start({version, options}, function watching(err) {
```
Comment (filmaj) [inline src/cli.js]: I think we want to make sure that if we're restarting the sandbox, that we're closing everything down properly. [`sandbox.start` provides a second argument to the callback which is a function to 'close everything down'](https://github.com/architect/sandbox/blob/master/src/start.js#L121) - I think we want to use that in the case of restarting.

File: src/cli.js [hunk 2]
```diff
@@ -1,14 +1,64 @@
...
+      if (event === 'update' &&
+          name.match(arcFile)) {
+        let indicator = chalk.green.dim('⚬')
+        let status = chalk.grey('Architect project manifest changed, reloading HTTP routes...')
+        console.log(`${indicator} ${status}`)
+
+        let start = Date.now()
+        process.env.QUIET = true
+        http.close()
+        http.start(function () {
```
Comment (filmaj) [inline src/cli.js]: From an abstraction point of view, I would be weary of mixing using the `sandbox.start` and `http.start` / `http.close` APIs in here. If `sandbox.start` changes how it uses `http`, but we don't update the usage of `http` in this file, then we could get our streams crossed up. I would consider modifying or adding to the `sandbox.start` API to enable the kind of interactions you are looking for here, _or_, restarting the entire sandbox using the `end()` function provided to the `start` callback, and calling up `sandbox.start` again.

**Comment by filmaj** [comment]
Oh, and once this functionality settles down, we should add tests for all of this 😊 happy to help with that!

File: src/cli.js (hunk 2 above)
Comment (ryanblock) [reply src/cli.js]: Yeah, as a separate exercise, related to this work we should probably establish a cleaner API into the various services sandbox starts and closes.

File: src/cli.js (hunk 1 above)
Comment (ryanblock) [reply src/cli.js]: This was intentional (for now); depending on your project size, shutting everything down can take quite a while. (Begin, for example, has a lot of tables, and can take a number of seconds to start up.) I'd like to move towards diffing the project upon change and restarting the specific service that needs to be restarted (as opposed to just http, or everything).

**Comment by ryanblock** [comment]
> Oh, and once this functionality settles down, we should add tests for all of this 😊 happy to help with that!

Broke out CLI tests, I can think of a few that could be added for sure!

**Pull request merged**